x-pack/plugins/enterprise_search/public/applications/app_search/index.tsx

```diff
@@ -18,18 +18,30 @@ import { EngineOverview } from './components/engine_overview';
 
 export const AppSearch: React.FC = () => {
   const { enterpriseSearchUrl } = useContext(KibanaContext) as IKibanaContext;
+  if (!enterpriseSearchUrl)
+    return (
+      <Switch>
+        <Route exact path="/setup_guide">
+          <SetupGuide />
+        </Route>
+        <Route>
+          <Redirect to="/setup_guide" />
+          <SetupGuide />
+        </Route>
+      </Switch>
+    );
 
   return (
     <Switch>
-      <Route exact path="/">
-        {!enterpriseSearchUrl ? <Redirect to="/setup_guide" /> : <Redirect to="/engines" />}
-      </Route>
       <Route exact path="/setup_guide">
         <SetupGuide />
       </Route>
       <Route>
         <Layout navigation={<AppSearchNav />}>
           <Switch>
+            <Route exact path="/">
+              <EngineOverview />
+            </Route>
```

Comment:
Seems like a bunch of duplicated code with the switch and setup guide routes. Could this be refactored?

Reply:
It could be, but I don't think it's super necessary. I've already spent several hours trying to wrangle React Router in e94f8ff - the `/` root redirects unfortunately behave differently in Kibana than it does in ent-search (I think because we inherit history from Kibana?) and I don't want to overcomplicate it. I'd rather be extremely explicit about what React Router is doing than try to be clever/hide code to save 5-10 lines 🤷 

That being said, maybe I'm not really fully seeing what refactor you mean? Feel free to give it a shot yourself if you want (just be very sure to test the `/setup_guide` URL and clicking the "App Search" breadcrumb back to the `/` index - that's where it tends to break)